app/src/controllers/object.js

```diff
@@ -43,6 +43,57 @@ const controller = {
     }
   },
 
+  /**
+   * @function addMetadata
+   * Creates a new version of the object via copy with the new metadata added
+   * @param {object} req Express request object
+   * @param {object} res Express response object
+   * @param {function} next The next callback function
+   * @returns {function} Express middleware function
+  */
+  async addMetadata(req, res, next) {
+    try {
+      const objId = addDashesToUuid(req.params.objId);
+      const objPath = getPath(objId)
+
+      const latest = await storageService.headObject({ filePath: objPath });
+      if (latest.ContentLength > 5368709120) {
+        throw new Error('Cannot copy an object larger than 5GB');
+      }
+      else {
+        if (Object.keys(req.query).length <= 0 ||
+          Object.keys(req.query).length === 1 && req.query.hasOwnProperty('versionId')) {
+          // 422 when no keys given
+          res.status(422).end();
+        }
+        else {
+          const metadataToAppend = req.query;
+
+          if (metadataToAppend.hasOwnProperty('versionId')) delete metadataToAppend['versionId'];
```

Comment:
It's bad practice to modify the request object stream like this. It's always better to instead construct a new object and copy out the subset of items you need from there instead. The way it's implemented right now, `metadataToAppend` is just a reference to the same `req.query` object, so if you deleted an attribute off from there, you're actually deleting it for both. Instead, you can achieve the same logic by using object destructuring to construct a new object.
```js
const { versionId, ...metadataToAppend } = req.query;
```